src/main.zig

```diff
@@ -266,6 +270,14 @@ const WlrSurface = struct {
     // --- wlroots Layer Shell ---
     /// The wlroots surface.
     wlr_surface: *zwlr.LayerSurfaceV1,
+    
+    /// Pending configuration from the compositor.
+    pending_config: ?struct {
+        serial: u32,
+        width: u32,
+        height: u32,
+    } = null,
+    config_dirty: bool = false,
```

Comment:
I like this change, but it should be its own commit, not squashed to a single commit. Please see the repo's commit history for examples on how fixes are committed.